src/main/java/emu/grasscutter/game/combine/CombineManger.java

```diff
@@ -12,15 +14,26 @@
 import emu.grasscutter.server.game.*;
 import emu.grasscutter.server.packet.send.*;
 import emu.grasscutter.utils.Utils;
+import io.netty.util.internal.ThreadLocalRandom;
 import it.unimi.dsi.fastutil.ints.*;
 import java.util.*;
 
 public class CombineManger extends BaseGameSystem {
     private static final Int2ObjectMap<List<Integer>> reliquaryDecomposeData =
             new Int2ObjectOpenHashMap<>();
+    private final Int2ObjectMap<CombineBonusData> combineBonusData = new Int2ObjectOpenHashMap<>();
 
     public CombineManger(GameServer server) {
         super(server);
+
+        // load combine bonus data
+        try {
+            DataLoader.loadList("CombineBonus.json", CombineBonusData.class)
+                    .forEach(entry -> combineBonusData.put(entry.getAvatarId(), entry));
+        } catch (Exception ignored) {
+            Grasscutter.getLogger()
```

Comment:
can we get a thing here to load the resource and copy it to the data dir? i dont expect people to copy the file when they see this warning